src/Weapon.ts

```diff
@@ -1 +1,82 @@
-// your code goes here
\ No newline at end of file
+// your code goes here
+import { Item } from './Item';
+
+export abstract class Weapon extends Item {
+  static readonly MODIFIER_CHANGE_RATE = 0.05;
+
+  private baseDamage: number;
+  private damageModifier: number;
+  private baseDurability: number;
+  private durabilityModifier: number;
+
+  constructor(name: string, baseDamage: number, baseDurability: number, value: number, weight: number) {
+    super(name, value, weight);
+
+    this.baseDamage = baseDamage;
+    this.validateDurability(baseDurability);
```

Comment:
It's convention to call any kind of validations before setting up any values to instance  props. 
In other words, just move ```this.validateDurability(baseDurability);``` on top of ```this.baseDamage = baseDamage;```